### ISSUE TITLE
When print order with Deliver date in the past - get error  data.shipments[0].options.delivery_date - Shipment validation error

### ISSUE THREAD

**Issue by magefan** [myparcelnl/magento#340]
**Describe the bug**
Fatal error

{"0":"Error in MyParcel API request: data.shipments[0].options.delivery_date - data.shipments[0].options.delivery_date - Shipment validation error Url: https:\/\/api.myparcel.nl\/shipments Request: {\"data\":{\"shipments\":[{\"recipient\":{\"cc\":\"BE\",\"person\":\"Roman Wr\\u00f6bel\",\"postal_code\":\"3630\",\"city\":\"Maasmechelen\",\"email\":\"roman.wrobel@hotmail.be\",\"phone\":\"0479270843\",\"company\":\"VMCP\",\"street\":\"Leeuwerikenstraat 29\",\"street_additional_info\":\"\"},\"options\":{\"package_type\":1,\"label_description\":\"2100013078\",\"only_recipient\":0,\"signature\":1,\"return\":0,\"delivery_type\":2,\"large_format\":0,\"delivery_date\":\"1970-01-03 00:00:00\"},\"carrier\":1,\"reference_identifier\":\"8698\",\"general_settings\":{\"disable_auto_detect_pickup\":0}}]}}","1":"<pre>#1 MyParcelNL\\Sdk\\src\\Helper\\MyParcelCollection->createConcepts() called at [vendor\/myparcelnl\/magento\/Model\/Sales\/MagentoOrderCollection.php:223]\n#2 MyParcelNL\\Magento\\Model\\Sales\\MagentoOrderCollection->createMyParcelConcepts() called at [vendor\/myparcelnl\/magento\/Controller\/Adminhtml\/Order\/CreateAndPrintMyParcelTrack.php:109]\n#3 MyParcelNL\\Magento\\Controller\\Adminhtml\\Order\\CreateAndPrintMyParcelTrack->massAction() called at [vendor\/myparcelnl\/magento\/Controller\/Adminhtml\/Order\/CreateAndPrintMyParcelTrack.php:59]\n#4 MyParcelNL\\Magento\\Controller\\Adminhtml\\Order\\CreateAndPrintMyParcelTrack->execute() called at [vendor\/magento\/framework\/App\/Action\/Action.php:108]\n#5 Magento\\Framework\\App\\Action\\Action->dispatch() called at [vendor\/magento\/framework\/Interception\/Interceptor.php:58]\n#6 MyParcelNL\\Magento\\Controller\\Adminhtml\\Order\\CreateAndPrintMyParcelTrack\\Interceptor->___callParent() called at [vendor\/magento\/framework\/Interception\/Interceptor.php:138]\n#7 MyParcelNL\\Magento\\Controller\\Adminhtml\\Order\\CreateAndPrintMyParcelTrack\\Interceptor->Magento\\Framework\\Interception\\{closure}() called at [vendor\/magento\/framework\/Interception\/Interceptor.php:153]\n#8 MyParcelNL\\Magento\\Controller\\Adminhtml\\Order\\CreateAndPrintMyParcelTrack\\Interceptor->___callPlugins() called at [generated\/code\/MyParcelNL\/Magento\/Controller\/Adminhtml\/Order\/CreateAndPrintMyParcelTrack\/Interceptor.php:26]\n#9 MyParcelNL\\Magento\\Controller\\Adminhtml\\Order\\CreateAndPrintMyParcelTrack\\Interceptor->dispatch() called at [vendor\/magento\/framework\/App\/FrontController.php:159]\n#10 Magento\\Framework\\App\\FrontController->processRequest() called at [vendor\/magento\/framework\/App\/FrontController.php:99]\n#11 Magento\\Framework\\App\\FrontController->dispatch() called at [vendor\/magento\/framework\/Interception\/Interceptor.php:58]\n#12 Magento\\Framework\\App\\FrontController\\Interceptor->___callParent() called at [vendor\/magento\/framework\/Interception\/Interceptor.php:138]\n#13 Magento\\Framework\\App\\FrontController\\Interceptor->Magento\\Framework\\Interception\\{closure}() called at [vendor\/magento\/framework\/Interception\/Interceptor.php:153]\n#14 Magento\\Framework\\App\\FrontController\\Interceptor->___callPlugins() called at [generated\/code\/Magento\/Framework\/App\/FrontController\/Interceptor.php:26]\n#15 Magento\\Framework\\App\\FrontController\\Interceptor->dispatch() called at [vendor\/magento\/framework\/App\/Http.php:137]\n#16 Magento\\Framework\\App\\Http->launch() called at [generated\/code\/Magento\/Framework\/App\/Http\/Interceptor.php:24]\n#17 Magento\\Framework\\App\\Http\\Interceptor->launch() called at [vendor\/magento\/framework\/App\/Bootstrap.php:261]\n#18 Magento\\Framework\\App\\Bootstrap->run() called at [pub\/index.php:44]\n<\/pre>","url":"\/xxxxxxx\/myparcelnl\/order\/CreateAndPrintMyParcelTrack\/key\/565e7d1f4b600d4e196837f6d2fa06b47975acc65fb43e3b6e742a50bcea7e3a\/","script_name":"\/index.php"}

**To Reproduce**
Place and order, and set the delivery date. In our case it was 18-12-2019 08:00 (https://prnt.sc/qczrdg), wait a few days, when this date expires, e.g. now is 19-12-2019. Go to print lable

**Expected behavior**
You get PDF file, but in fact, there is an error page, with the error I mentioned before

**Desktop (please complete the following information):**
 - OS: any
 - Browser any
 - MyParcel Plugin version - 3.1.4-beta.2
 - MyParcel SDK version - v3.1.2-beta.1

**Additional context**
As a temporary solution, we made this change in file
vendor/myparcelnl/sdk/src/Helper/MyParcelCurl.php in function createConcepts, before $request = (new MyParcelRequest()) put this code

```
/* Custom fix */
            $data = json_decode($data, true);
            if (isset($data['data']['shipments'][0]['options']['delivery_date'])
               && $data['data']['shipments'][0]['options']['delivery_date'] < date('Y-m-d H:i:s')
            ) {
               $data['data']['shipments'][0]['options']['delivery_date'] = date('Y-m-d H:i:s', time() + 86400 * 2); 
            }
            $data = json_encode($data);
            /* Custom fix end */

```

We don't think that this is the best solution. What I suggest is to catch this error, and display some useful admin user error and allow to change the delivery date in admin panel   https://prnt.sc/qczrdg

**Comment by TeunSmit** [comment]
684176

**Comment by magefan** [comment]
Also you 100% need to add try-catch to the controllers when error happened , e.g
MyParcelNL\Magento\Controller\Adminhtml\Order\CreateAndPrintMyParcelTrack make execute
Pull request:
https://github.com/myparcelnl/magento/pull/343/files

**Comment by RichardPerdaan** [comment]
Problem fixed in version [3.2.0-beta.1](https://github.com/myparcelnl/magento/releases/tag/3.2.0-beta.1)
